ouroboros-consensus/docs/future-ledger-hd.md

```diff
@@ -0,0 +1,219 @@
+# Ledger-HD sketch (UTxO-HD v2)
+
+This document describes the result of the discussion between Ledger and
+Consensus teams on 2022-01-17 about the future steps on UTxO-HD which would not
+make sense to be called this way anymore in future versions and therefore we
+propose Ledger-HD as a replacement.
+
+Below, we outline the tables that are expected to be moved to the disk, their
+dynamics, the main computations that will leave the Ledger and some open
+questions.
+
+This is meant to be just a sketch, details are not worked out yet.
+
+## Scope of Ledger-HD
+
+The plan for Ledger-HD is to move the following tables to the disk:
+
+- The unified map of `rewards`, `delegations`, `pointers` and `deposits`
+
+```haskell
+( ne :: NewEpochState era )
+  & ( nesEs     :: NewEpochState      era  -> EpochState         era  )
+  & ( esLState  :: EpochState         era  -> LedgerState        era  )
+  & ( lsDPState :: LedgerState        era  -> DPState (EraCrypto era) )
+  & ( dpsDState :: DPState (EraCrypto era) -> DState  (EraCrypto era) )
+  & ( dsUnified :: DState  (EraCrypto era) -> UMap    (EraCrypto era) )
+   
+data UMap c = UMap !(Map (Credential 'Staking c) (Trip c)) !(Map Ptr (Credential 'Staking c)) 
+```
+
+> **_NOTE:_** The `deposits` don't seem to be in-place yet? For me it still is a
+> TriMap
```

Comment:
it is there in the `master` branch now, maybe you have a older version?

Reply:
That would explain it, yeah :smile: